recommendation/microsoft.owin.json

```diff
@@ -0,0 +1,352 @@
+            {
+              "Name": "ReplaceIdentifier",
+              "Type": "Identifier",
+              "Value": "HttpContext",
+              "Description": "Replace IOwinContext with HttpContext",
+              "ActionValidation": {
+                "Contains": "HttpContext",
+                "NotContains": "IOwinContext"
+              }
+            },
+            {
+              "Name": "AddDirective",
+              "Type": "Using",
+              "Value": "Microsoft.AspNetCore.Http",
+              "Description": "Add Microsoft.AspNetCore.Http namespace",
+              "ActionValidation": {
+                "Contains": "usingMicrosoft.AspNetCore.Http",
+                "NotContains": ""
+              }
+            }
+          ]
+        }
+      ]
+    },
+    {
+      "Type": "Class",
+      "Name": "OwinMiddleware",
+      "Value": "Microsoft.Owin.OwinMiddleware",
+      "KeyType": "BaseClass",
+      "ContainingType": "",
+      "RecommendedActions": [
+        {
+          "Source": "Amazon",
+          "Preferred": "Yes",
+          "TargetFrameworks": [
+            {
+              "Name": "netcoreapp3.1",
+              "TargetCPU": [ "x86", "x64", "ARM32", "ARM64" ]
+            },
+            {
+              "Name": "net5.0",
+              "TargetCPU": [ "x86", "x64", "ARM32", "ARM64" ]
+            }
+          ],
+          "Description": "Removes base class of OwinMiddleware\r\n",
+          "Actions": [
+            {
+              "Name": "RemoveBaseClass",
+              "Type": "Class",
+              "Value": "OwinMiddleware",
+              "Description": "Removes base class OwinMiddleware",
+              "ActionValidation": {
+                "Contains": "",
+                "NotContains": "OwinMiddleware"
+              }
+            }
+          ]
+        }
+      ]
+    },
+    {
+      "Type": "Class",
+      "Name": "OwinMiddleware",
+      "Value": "Microsoft.Owin.OwinMiddleware",
+      "KeyType": "Identifier",
+      "ContainingType": "",
+      "RecommendedActions": [
+        {
+          "Source": "Amazon",
+          "Preferred": "Yes",
+          "TargetFrameworks": [
+            {
+              "Name": "netcoreapp3.1",
+              "TargetCPU": [ "x86", "x64", "ARM32", "ARM64" ]
+            },
+            {
+              "Name": "net5.0",
+              "TargetCPU": [ "x86", "x64", "ARM32", "ARM64" ]
+            }
+          ],
+          "Description": "Replace OwinMiddleware with RequestDelegate\r\nAdd Microsoft.AspNetCore.Http.RequestDelegate namespace\r\nRemove Microsoft.Owin.OwinMiddleware\r\n",
+          "Actions": [
+            {
+              "Name": "ReplaceIdentifier",
+              "Type": "Identifier",
+              "Value": "RequestDelegate",
+              "Description": "Replace OwinMiddleware with RequestDelegate",
+              "ActionValidation": {
+                "Contains": "RequestDelegate",
+                "NotContains": "OwinMiddleware"
```

Comment:
Do these have the same APIs when replaced?

Reply:
Discussed with Mark, and this is a valid approach for replace OwinMiddleware abstract class.